test/Table_Tests/src/Common_Table_Operations/Column_Operations_Spec.enso

```diff
@@ -827,6 +827,21 @@ spec setup =
                 s1.like (s2+"%r") . to_vector . should_equal [True, False, False, False, Nothing]
                 s1.like "%r%" . to_vector . should_equal [True, True, False, False, Nothing]
 
+        Test.specify "should handle operation text_length" <|
+            t = table_builder [["strings", ["foobar", "", Nothing, "👩‍🔬", "café", "It was the best of times, it was the worst of times, it was the age of wisdom, it was the age of foolishness, it was the epoch of belief, it was the epoch of incredulity, it was the season of light, it was the season of darkness, it was the spring of hope, it was the winter of despair."]]]
+            col = t.at "strings"
+            res = col.text_length
+            res.name . should_equal "text_length([strings])"
+
+            case setup.is_database of
+                False -> res . to_vector . should_equal [6, 0, Nothing, 1, 4, 286] # Grapheme Length
+                True -> res . to_vector . should_equal [6, 0, Nothing, 3, 4, 286]  # Storage Length
+
+        Test.specify "text_length should error on non-string columns" <|
+            t = table_builder [["numbers", [1, 2, 3]]]
+            col = t.at "numbers"
+            col.text_length . should_fail_with Invalid_Value_Type
+
         Test.specify "should handle operations like is_empty, is_blank, fill_empty" <|
             with_mixed_columns_if_supported [["s", ["", " ", "  ", Nothing, "foo"]], ["letters", ["a", "b", "c", "d", "e"]]] t->
```

Comment:
The test looks perfect.

A suggestion that it may be worth to use the `with_mixed_columns_if_supported` feature here as well.

We have some logic that even if a column is of mixed type (`col.value_type == Mixed`) but contains all elements of a given type (e.g. text), then the `expect_text` operation will accept it (because it actually relies on `inferred_precise_value_type`). Thus it is good to ensure that the operation still works correctly in that scenario (I don't think there is any reason why it wouldn't work, but that's what the tests are for after all 🙂).

Reply:
Done and still passes